ui/app/services/api.ts

```diff
@@ -103,27 +104,6 @@ export default class ApiService extends Service {
       this.controlGroup.deleteControlGroupToken(controlGroupToken.accessor);
     }
   };
-
-  formatErrorResponse = async (context: ResponseContext) => {
-    const response = context.response.clone();
-    const { headers, status, statusText } = response;
-
-    // backwards compatibility with Ember Data
-    if (status >= 400) {
-      const error: ApiError = (await response?.json()) || {};
-      error.httpStatus = response?.status;
-      error.path = context.url;
-      // typically the Vault API error response looks like { errors: ['some error message'] }
-      // but sometimes (eg RespondWithStatusCode) it's { data: { error: 'some error message' } }
-      if (error?.data?.error && !error.errors) {
-        // normalize the errors from RespondWithStatusCode
-        error.errors = [error.data.error];
-      }
-      return new Response(JSON.stringify(error), { headers, status, statusText });
-    }
-
-    return;
-  };
```

Comment:
This middleware isn't useful since all error handling should go through the `parseError` method and I moved the `path` and `error.errors` parsing there. The `httpStatus` was a shim for Ember Data but it's unnecessary and will be updated where referenced to use the status returned from the `parseError` method.